bake/remote.go

```diff
@@ -23,13 +25,42 @@ type Input struct {
 }
 
 func ReadRemoteFiles(ctx context.Context, nodes []builder.Node, url string, names []string, pw progress.Writer) ([]File, *Input, error) {
-	var session []session.Attachable
+	var sessions []session.Attachable
 	var filename string
+
 	st, ok := dockerui.DetectGitContext(url, false)
 	if ok {
-		ssh, err := controllerapi.CreateSSH([]*controllerapi.SSH{{ID: "default"}})
-		if err == nil {
-			session = append(session, ssh)
+		var sshSpecs []*controllerapi.SSH
+		if v := os.Getenv("BUILDX_BAKE_GIT_SSH"); v != "" {
+			var err error
+			sshSpecs, err = buildflags.ParseSSHSpecs([]string{v})
+			if err != nil {
+				return nil, nil, errors.Wrapf(err, "failed to parse BUILDX_BAKE_GIT_SSH")
+			}
+		} else {
+			sshSpecs = []*controllerapi.SSH{{ID: "default"}}
+		}
+		if ssh, err := controllerapi.CreateSSH(sshSpecs); err == nil {
+			sessions = append(sessions, ssh)
+		}
+
+		var gitAuthSecrets []*controllerapi.Secret
+		if _, ok := os.LookupEnv("BUILDX_BAKE_GIT_AUTH_TOKEN"); ok {
+			gitAuthSecrets = append(gitAuthSecrets, &controllerapi.Secret{
+				ID:  "GIT_AUTH_TOKEN",
```

Comment:
we should expose this key in buildkit to avoid hardcoded value. Same for `GIT_AUTH_HEADER`

Reply:
opened https://github.com/moby/buildkit/pull/4809